### PR TITLE
[Separate Reporter Part 5] Implement deviation reporter

### PR DESCRIPTION
**Pull request by Intizar-T**
# Description

Please include a summary of the changes and the related issue.
Please also include relevant motivation and context.
List any dependencies that are required for this change.

Related issue #1777 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Deployment

- [ ] Should publish npm package
- [ ] Should publish Docker image

### CODE REVIEW

File: node/pkg/reporter/reporter.go [hunk 1]
```diff
@@ -37,6 +35,7 @@ func NewReporter(ctx context.Context, opts ...ReporterOption) (*Reporter, error)
 		SubmissionInterval: groupInterval,
 		CachedWhitelist:    config.CachedWhitelist,
 		deviationThreshold: deviationThreshold,
+		JobType:            config.JobType,
```
Comment (nick-bisonai) [inline node/pkg/reporter/reporter.go:38]: Let's remove this JobType and declare function property instead, I think it'll remove if branch and make the code more simple

```
type Reporter struct {
	KaiaHelper         *helper.ChainHelper
	SubmissionPairs    map[int32]SubmissionPair
	SubmissionInterval time.Duration
	CachedWhitelist    []common.Address

	contractAddress    string
	deviationThreshold float64

	nodeCtx    context.Context
	nodeCancel context.CancelFunc
	isRunning  bool

	LatestData *sync.Map // map[symbol]SubmissionData

	Job func() err
}
```

**Review by nick-bisonai — APPROVED** [review]
lgtm! left few comments

**Comment by nick-bisonai** [comment]
@coderabbitai review

File: node/pkg/reporter/app.go [hunk 2]
```diff
@@ -78,78 +78,46 @@ func (a *App) setReporters(ctx context.Context) error {
 			WithInterval(groupInterval),
 			WithContractAddress(contractAddress),
 			WithCachedWhitelist(cachedWhitelist),
+			WithKaiaHelper(chainHelper),
+			WithLatestData(&a.LatestData),
```
Comment (nick-bisonai) [inline node/pkg/reporter/app.go:82]: let's define app.LatestData as *sync.Map type, and allocate memory from App.New() func. then, we can remove `&` here

Comment (Intizar-T) [reply node/pkg/reporter/app.go:82]: done! will be reflected in the next PR

**Pull request merged**